packages/puppeteer-core/src/bidi/ExposedFunction.ts

```diff
@@ -38,65 +32,50 @@ export class ExposeableFunction<Args extends unknown[], Ret> {
 
   readonly name;
   readonly #apply;
+  readonly #isolate;
 
-  readonly #channels;
-  readonly #callerInfos = new Map<
-    string,
-    Map<number, RemotePromiseCallbacks>
-  >();
+  readonly #channel;
 
-  #preloadScriptId?: Bidi.Script.PreloadScript;
+  #scripts: Array<[BidiFrame, Bidi.Script.PreloadScript]> = [];
 
   constructor(
     frame: BidiFrame,
     name: string,
-    apply: (...args: Args) => Awaitable<Ret>
+    apply: (...args: Args) => Awaitable<Ret>,
+    isolate = false
   ) {
     this.#frame = frame;
     this.name = name;
     this.#apply = apply;
+    this.#isolate = isolate;
 
-    this.#channels = {
-      args: `__puppeteer__${this.#frame._id}_page_exposeFunction_${this.name}_args`,
-      resolve: `__puppeteer__${this.#frame._id}_page_exposeFunction_${this.name}_resolve`,
-      reject: `__puppeteer__${this.#frame._id}_page_exposeFunction_${this.name}_reject`,
-    };
+    this.#channel = `__puppeteer__${this.#frame._id}_page_exposeFunction_${this.name}`;
   }
 
   async expose(): Promise<void> {
     const connection = this.#connection;
-    const channelArguments = this.#channelArguments;
+    const channel = {
+      type: 'channel' as const,
+      value: {
+        channel: this.#channel,
+        ownership: Bidi.Script.ResultOwnership.Root,
+      },
+    };
 
     // TODO(jrandolf): Implement cleanup with removePreloadScript.
```

Comment:
implemented now?

Reply:
This is for `removeExposeFunction`

Reply:
I've removed the comment. Linking the bug here instead.